src/main/java/eu/darkbot/api/entities/Portal.java

```diff
@@ -0,0 +1,26 @@
+package eu.darkbot.api.entities;
+
+import eu.darkbot.api.entities.utils.Map;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represent in-game portal {@link Entity}
+ */
+public interface Portal extends Entity {
+
+    /**
+     * @return destination {@link Map} of this {@link Portal}
+     */
+    @Nullable
+    Map getTargetMap();
+
+    /**
+     * Portal type id in generally is id of the portal's graphic.
+     *
+     * @return {@link Portal}'s type id
+     */
+    int getPortalType();
+
+    // TODO: 01.11.2020 not sure which and when.
+    int getFactionId();
```

Comment:
5-1, 5-2 and 5-3 portals, are from a specific faction, and members from other factions don't see them on the map